docs: update design/tradeoffs doc regarding scanner vuln reports #818 (#854)
project-copacetic/copacetic: website/docs/design.md

## Tradeoffs

The core architectural choice of using packages as the unit of patching creates several constraints:

1. **Package-Based Update Model**:

>    - While Copa can work with or without vulnerability scanner reports, its patching capability is fundamentally based on os-level package updates

Reviewer: ```suggestion
   - While Copa can work with or without vulnerability scanner reports, its patching capability is fundamentally based on package updates
```

   - When using scanner reports to perform a targeted patching of packages, false positives/negatives from scanners flow downstream to Copa

2. **Package Manager Dependencies**:
   - Copa depends on individual package manager adapters to correctly deploy patches to target images
   - There is a long tail of compatibility issues that arise depending on the target image itself:
     - Outdated package manager config/keys
     - Invalid/missing package graph
     - Repository configuration issues
   - Overall, the maintenance cost of the project is expected to be non-trivial to address these package manager complexities

3. **Platform Limitations**:

>    - No support for Windows containers given the dependency on buildkit

Reviewer: although, we haven't done much exploration, buildkit does support buildkit experimentally now. however, windows containers themselves doesn't have an upgrade mechanism afaik.

Author: Covered.

## Tradeoffs

> The core architectural choice of using packages as the unit of patching creates several constraints:

Reviewer 2: we can remove this sentence since it is repetitive and the third tradeoff wouldn't fall under this

Reviewer 2: @pradhans0906 thank you for the changes! just had this additional comment then LGTM

1. **Package-Based Update Model**:

>    - While copa can work with or without vulnerability scanner reports, its patching capability is fundamentally based on package updates
>    - When using scanner reports, false positives/negatives from scanners flow downstream to copa
>    - The package-based approach limits the kinds of vulnerabilities copa can address to those fixable via package updates

Reviewer 2: ```suggestion
   - While Copa can work with or without vulnerability scanner reports, its patching capability is fundamentally based on os-level package updates
   - When using scanner reports to perform a targeted patching of packages, false positives/negatives from scanners flow downstream to Copa
```

2. **Package Manager Dependencies**:
   - Copa depends on individual package manager adapters to correctly deploy patches to target images
   - There is a long tail of compatibility issues that arise depending on the target image itself:
     - Outdated package manager config/keys
     - Invalid/missing package graph
     - Repository configuration issues
   - Overall, the maintenance cost of the project is expected to be non-trivial to address these package manager complexities

3. **Platform Limitations**:
   - No support for Windows containers given the dependency on buildkit

> This design prioritizes practicality and integration with existing Linux package management ecosystems, while accepting the inherent limitations and maintenance overhead of this approach.

Reviewer 2: ```suggestion
```
